topgun/k8s/k8s_suite_test.go

```diff
@@ -307,30 +326,16 @@ func helmDestroy(releaseName string) {
 	Wait(Start(nil, "helm", helmArgs...))
 }
 
-func getPods(namespace string, flags ...string) []pod {
-	var (
-		pods struct {
-			Items []pod `json:"items"`
-		}
-
-		args = append([]string{"get", "pods",
-			"--namespace=" + namespace,
-			"--output=json",
-			"--no-headers"}, flags...)
-		session = Start(nil, "kubectl", args...)
-	)
-
-	Wait(session)
-
-	err := json.Unmarshal(session.Out.Contents(), &pods)
+func getPods(namespace string, listOptions metav1.ListOptions) []corev1.Pod {
+	pods, err := kubeClient.CoreV1().Pods(namespace).List(listOptions)
```

Comment:
yayyyy

Reply:
aaah actually, seems like we're getting quite a lot of nil pointer derefs here (40 :grimacing: )

```
  Test Panicked
  runtime error: invalid memory address or nil pointer dereference
  /usr/local/go/src/runtime/panic.go:212

  Full Stack Trace
  k8s.io/client-go/kubernetes.(*Clientset).CoreV1(...)
        /home/ubuntu/go/pkg/mod/k8s.io/client-go@v11.0.0+incompatible/kubernetes/clientset.go:237
  github.com/concourse/concourse/topgun/k8s_test.getPods(0xc0001277e0, 0x12, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/ubuntu/workspace/concourse/topgun/k8s/k8s_suite_test.go:330 +0x3f
  github.com/concourse/concourse/topgun/k8s_test.waitAllPodsInNamespaceToBeReady.func1(0x0)
        /home/ubuntu/workspace/concourse/topgun/k8s/k8s_suite_test.go:350 +0x83
  reflect.Value.call(0x1162be0, 0xc0005be020, 0x13, 0x132b124, 0x4, 0xc00054f070, 0x0, 0x0, 0x1162be0, 0x1, ...)
        /usr/local/go/src/reflect/value.go:460 +0x8ab
  reflect.Value.Call(0x1162be0, 0xc0005be020, 0x13, 0xc00054f070, 0x0, 0x0, 0x1, 0x1, 0x1)
        /usr/local/go/src/reflect/value.go:321 +0xb4
  github.com/onsi/gomega/internal/asyncassertion.(*AsyncAssertion).pollActual(0xc0004eca80, 0xc000441ae0, 0x1f, 0x132a756, 0x1)
        /home/ubuntu/go/pkg/mod/github.com/onsi/gomega@v1.7.1/internal/asyncassertion/async_assertion.go:75 +0xe9
  github.com/onsi/gomega/internal/asyncassertion.(*AsyncAssertion).match(0xc0004eca80, 0x14f58c0, 0x1e807e8, 0x40ea01, 0xc000490150, 0x1, 0x1, 0xc000490150)
```

two example failures are

```
[Panic!] external workers through separate deployments [JustBeforeEach] generic worker web with correct public key worker registers with atc
/usr/local/go/src/runtime/panic.go:212

[Panic!] external workers through separate deployments [JustBeforeEach] generic worker web with invalid public key worker doesn't register
/usr/local/go/src/runtime/panic.go:212
```